### PR TITLE
docs: update getting started topics to avoid duplicate topic names

### LINKED ISSUE #35456: Updating getting started topics to avoid duplicate topic names
# 📚 Docs or angular.io bug report

### Description

In our documentation, we have two deployment.md files. One in the Getting Started section, the other in the rest of the Guide.

This can be confusing to users. To fix this, we should:

Rename all getting started topics to prepend `start-` to the filename.

Retitle all getting started topics to have "Getting started with Angular: " (same as the first topic).

Update navigate.json accordingly.

### What's the affected URL?**
angular.io/docs/start

### Reproduction Steps**
Search for Angular deployment

### Expected vs Actual Behavior**
You could get two topics: one for getting started, the other more general.

## 📷Screenshot
N/A

## 🔥 Exception or Error
N/A

### Browser info
N/A

### Anything else relevant?
Nope!

### PR DESCRIPTION
**Pull request by aikithoughts**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #35456

## What is the new behavior?
Renamed start topics to prevent duplication. Updated titles.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Resolves #35456

### CODE REVIEW

**Comment by kapunahelewong** [comment]
Regarding off-line conversation about redirects, here is the link for reference: https://github.com/angular/angular/blob/master/aio/firebase.json

File: aio/firebase.json [hunk 1]
```diff
@@ -30,6 +30,12 @@
       {"type": 301, "source": "/getting-started", "destination": "/start"},
       {"type": 301, "source": "/getting-started/:rest*", "destination": "/start/:rest*"},
 
+      // Renaming of Getting Started topics
+      {"type": 301, "source": "/start/deployment", "destination": "/start/start-deployment"},
+      {"type": 301, "source": "/start/data", "destination": "/start/start-data"},
```
Comment (gkalpak) [inline aio/firebase.json]: You could flip these two lines and have alphabetic order :tada:

File: aio/ngsw-config.json [hunk 2]
```diff
@@ -128,6 +128,18 @@
...
+    "!/start/data.html",
+    "!/start/deployment",
+    "!/start/deployment/",
+    "!/start/deployment.html",
+    "!/start/forms",
+    "!/start/forms/",
+    "!/start/forms.html",
+    "!/start/routing",
+    "!/start/routing/",
+    "!/start/routing.html",
```
Comment (gkalpak) [inline aio/ngsw-config.json]: Maybe move these below `!/news...` for alphabetic order 🙏

File: aio/src/index.html [hunk 3]
```diff
@@ -95,7 +95,7 @@
       <div class="hero-logo">
         <img src="assets/images/logos/angular/angular.svg" width="250" height="250" alt="Angular">
       </div>
-      <div class="homepage-container">
+      <div class="c">
```
Comment (gkalpak) [inline aio/src/index.html]: 🤔

**Review by gkalpak — COMMENTED** [review]
Nice 👌 

BTW, you also need to update [tests/deployment/shared/URLS_TO_REDIRECT.txt][1] with the correct destination URL and also add some new entries for the new redirects (i.e. `/start/data` --> `/start/start-data`, etc.).

[1]: https://github.com/angular/angular/blob/1997b86a0/aio/tests/deployment/shared/URLS_TO_REDIRECT.txt#L175-L178

File: aio/firebase.json (hunk 1 above)
Comment (aikidave) [reply aio/firebase.json]: Done!

File: aio/ngsw-config.json (hunk 2 above)
Comment (aikidave) [reply aio/ngsw-config.json]: Done!

File: aio/src/index.html (hunk 3 above)
Comment (aikidave) [reply aio/src/index.html]: Odd! I've reverted this file.

File: aio/tests/deployment/shared/URLS_TO_REDIRECT.txt [hunk 4]
```diff
@@ -187,5 +187,9 @@
 /guide/webpack	https://v5.angular.io/guide/webpack
 /news	https://blog.angular.io/
 /news.html	https://blog.angular.io/
+/start/data  /start/start-data
+/start/deployment  /start/start-deployment
+/start/forms  /start/start-forms
+/start/routing  /start/start-routing
```
Comment (gkalpak) [inline aio/tests/deployment/shared/URLS_TO_REDIRECT.txt]: :+1:

BTW, you also need to fix the `/getting-started/...` redirects above.

**Review by gkalpak — APPROVED** [review]
LGTM for docs-infra (with my one comment addressed) as soon as CI is happy.

**Comment by Splaktar** [comment]
Screen Shot 2020-02-24 at 17 23 51

It seems like "Getting Started with Angular: Your First App" should be under "START" instead of "OTHER"?

File: aio/tests/deployment/shared/URLS_TO_REDIRECT.txt [hunk 5]
```diff
@@ -172,10 +172,10 @@
...
 /docs/ts/latest/cookbook/ts-to-js.html	https://v2.angular.io/docs/ts/latest/cookbook/ts-to-js.html
 /getting-started	/start
-/getting-started/routing	/start/routing
-/getting-started/data	/start/data
-/getting-started/forms	/start/forms
-/getting-started/deployment	/start/deployment
+/getting-started/routing	/start/start-routing
+/getting-started/data	/start/start-data
+/getting-started/forms	/start/start-forms
+/getting-started/deployment	/start/start-deployment
```
Comment (Splaktar) [inline aio/tests/deployment/shared/URLS_TO_REDIRECT.txt]: Are these already handled in `aio/firebase.json` by https://github.com/angular/angular/pull/35457/files#diff-45a3893504fac74c45720da147e7c77fR31 or is something more needed there as well?

Comment (aikidave) [reply aio/tests/deployment/shared/URLS_TO_REDIRECT.txt]: I've been following what @gkalpak has asked me to do with regards to these redirects. Hoping we can clearly document how to do redirects going forward!

**Comment by aikidave** [comment]
I don't know how search results are handled in Angular.io, so I can't help with that part...yet. Is that something worth blocking this PR though?

**Comment by Splaktar** [comment]
> I don't know how search results are handled in Angular.io, so I can't help with that part...yet. Is that something worth blocking this PR though?

If caused by this PR, then yes, but that's not clear. @petebacondarwin any input on this?

File: aio/tests/deployment/shared/URLS_TO_REDIRECT.txt (hunk 5 above)
Comment (Splaktar) [reply aio/tests/deployment/shared/URLS_TO_REDIRECT.txt]: Yep, the question was mostly directed at him.

**Comment by gkalpak** [comment]
It is not caused by this PR, so we are good (for this one). It can be fixed in another PR :wink:

File: aio/tests/deployment/shared/URLS_TO_REDIRECT.txt (hunk 5 above)
Comment (gkalpak) [reply aio/tests/deployment/shared/URLS_TO_REDIRECT.txt]: They are handled in two steps: 
1. Redirecting `/getting-started/<something>` to `/start/<something>` ([here][1]).
2. Redirecting `/start/<something>` to `/start/start-<something>` ([there][2]).

So, nothing more is needed.

[1]: https://github.com/angular/angular/pull/35457/files#diff-45a3893504fac74c45720da147e7c77fR31
[2]: https://github.com/angular/angular/pull/35457/files#diff-45a3893504fac74c45720da147e7c77fR34-R37

**Review by kapunahelewong — APPROVED** [review]

**Review by IgorMinar — APPROVED** [review]
lgtm, thanks!!

**Comment by Splaktar** [comment]
OK, I've opened https://github.com/angular/angular/issues/35676 to track the issue with the search results.

**Pull request closed without merging**